tests/modules/commerce_redirect_test/src/Form/RedirectTestSubForm.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Drupal\commerce_redirect_test\Form;
+
+use Drupal\commerce\Response\NeedsRedirectException;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Tests throwing a redirect inside of a form.
+ */
+class RedirectTestSubForm extends FormBase {
```

Comment:
I think this form is unneeded, because it doesn't reflect how we do subforms (we don't use form builder). Better to do the exception in the parent.